scripts/verify-claimed-stake-fees.ts

```diff
@@ -0,0 +1,206 @@
+import {ethers} from 'hardhat';
+import hre from 'hardhat';
+import {EthersProviderWrapper} from '../deploy/ethers-provider-wrapper';
+import {DelegatorPool, L2Migrator} from '../typechain';
+import {BigNumber} from 'ethers';
+
+const bondingManagerABI = [
+  'function pendingStake(address _delegator, uint256 _endRound) public view returns (uint256)',
+  'function pendingFees(address _delegator, uint256 _endRound) public view returns (uint256)',
+];
+
+const getL1Pending = async (orchAddr: string) => {
+  const bondingManagerAddr = '0x511bc4556d823ae99630ae8de28b9b80df90ea2e';
+  const roundNum = 2466;
+
+  const bondingManager = new ethers.Contract(
+      bondingManagerAddr,
+      bondingManagerABI,
+      new EthersProviderWrapper(hre.companionNetworks['l1'].provider),
+  );
+
+  const [pendingStake, pendingFees] = await Promise.all([
+    await bondingManager.pendingStake(orchAddr, roundNum),
+    await bondingManager.pendingFees(orchAddr, roundNum),
+  ]);
+
+  return {
+    address: orchAddr,
+    stake: pendingStake,
+    fees: pendingFees,
+  };
+};
+
+const getL2Pending = async (orchAddr: string, claimBlock: number) => {
+  const bondingManagerAddr = '0x35Bcf3c30594191d53231E4FF333E8A770453e40';
+  const roundNum = 2466; // does not matter on L2, only current round is used
+
+  const bondingManager = new ethers.Contract(
+      bondingManagerAddr,
+      bondingManagerABI,
+      new EthersProviderWrapper(hre.network.provider),
+  );
+
+  const [pendingStake, pendingFees] = await Promise.all([
+    await bondingManager.pendingStake(orchAddr, roundNum, {
+      blockTag: claimBlock,
+    }),
+    await bondingManager.pendingFees(orchAddr, roundNum, {
+      blockTag: claimBlock,
+    }),
+  ]);
+
+  return {
+    address: orchAddr,
+    stake: pendingStake,
+    fees: pendingFees,
+  };
+};
+
+const getClaimers = async () => {
+  const l2MigratorAddr = (await hre.deployments.get('L2MigratorProxy')).address;
+  const claimStakeEnableBlock = 6737210;
+
+  const l2Migrator: L2Migrator = await ethers.getContractAt(
+      'L2Migrator',
+      l2MigratorAddr,
+  );
+
+  const events = await l2Migrator.queryFilter(
+      l2Migrator.filters.StakeClaimed(),
+      claimStakeEnableBlock,
+      'latest',
+  );
+
+  console.log(events.length, 'delegators claimed stake and fees');
+
+  const data: {
+    address: string;
+    claimBlock: number;
+    requestedStake: BigNumber;
+    requestedFees: BigNumber;
+    owedStake: BigNumber;
+    owedFees: BigNumber;
+  }[] = [];
+
+  // faster - run if order of addresses not important
```

Comment:
[optional] perhaps for readability's sake we can remove the commented out code from this script, rename the script to ```verify-unsorted-claimed-stake-fees.ts``` and have another script called ```verify-sorted-claimed-stake-fees.ts```

Reply:
added sorting with parallel promise resolution and removed sequential promise resolution 3e3e0dc